util/patch/patch.go

```diff
@@ -146,8 +146,11 @@ func (h *Helper) Patch(ctx context.Context, obj client.Object, opts ...Option) e
 	if err := h.patch(ctx, obj); err != nil {
 		errs = append(errs, err)
 	}
+
 	if err := h.patchStatus(ctx, obj); err != nil {
-		errs = append(errs, err)
+		if !(apierrors.IsNotFound(err) && !obj.GetDeletionTimestamp().IsZero() && len(obj.GetFinalizers()) == 0) {
```

Comment:
Should we add a test?

Reply:
I can add a test for this and submit a new PR.

Reply:
PR with the test https://github.com/kubernetes-sigs/cluster-api/pull/10866